### PR TITLE
Replacing "new" operators with make_unique/shared

### PR DESCRIPTION
**Pull request by ShFil119**
I've found some snippets of code to edit in c++ 11 "style".
I've tried to keep readability as good as possible.

### CODE REVIEW

File: rwengine/src/BinaryStream.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ std::unique_ptr<BinaryStream> BinaryStream::parse(const std::string &filename) {
     dfile.read(data, length);
     // std::cout << "File is " << length << " bytes" << std::endl << std::endl;
 
-    auto BS = std::unique_ptr<BinaryStream>(new BinaryStream);
+    std::unique_ptr<BinaryStream> BS = std::make_unique<BinaryStream>();
```
Comment (paulsapps) [inline rwengine/src/BinaryStream.cpp]: might as well be `auto BS = std::make_unique<BinaryStream>();`

Comment (paulsapps) [inline rwengine/src/BinaryStream.cpp]: Also there is still `char *data = new char[length];` in here which could be vector<char>

File: rwengine/src/loaders/LoaderIDE.cpp [hunk 2]
```diff
@@ -65,8 +65,7 @@ bool LoaderIDE::load(const std::string &filename, const PedStatsList &stats) {
                     break;
                 case OBJS:
                 case TOBJ: {  // Supports Type 1, 2 and 3
-                    auto objs =
-                        std::unique_ptr<SimpleModelInfo>(new SimpleModelInfo);
+                    std::unique_ptr<SimpleModelInfo> objs = std::make_unique<SimpleModelInfo>();
```
Comment (paulsapps) [inline rwengine/src/loaders/LoaderIDE.cpp]: In this switch case nothing happens with the result obj... can't see why it needs to be on the heap?

**Comment by ShFil119** [comment]
Do you prefer new pull, or rewrite commits?

**Comment by danhedron** [comment]
Preferably rewrite commits unless the entire PR changes

**Comment by ShFil119** [comment]
Done. ;)
"Have you run clang-format on the files? Some lines look like they're a bit longer than before."
No, but can you give a detail what/how change?

**Pull request merged**